catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/BackupCommandTest.java

```diff
@@ -0,0 +1,976 @@
+        backupStatusCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(), containsString("Backup status failed."));
+    }
+
+    /**
+     * Collections are optimized before backups. This test verifies that an error message is
+     * printed to the console when optimization of a collection fails.
+     */
+    @Test
+    public void testSolrCloudBackupFailsDuringOptimizationWithErrorCode() throws Exception {
+
+        // Set the solr client type system property so that the
+        // BackupCommand knows that it needs to backup solr cloud.
+        setupSolrClientType(SOLR_CLOUD_CLIENT_TYPE);
+
+        setupMockSolrClientForCollectionOptimization(DEFAULT_CORE_NAME, FAILURE_STATUS_CODE);
+
+        BackupCommand backupCommand = getSynchronousBackupCommand(getBackupLocation(),
+                DEFAULT_CORE_NAME,
+                mockSolrClient);
+
+        // Peform sync backup
+        backupCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format(
+                        "Backing up collection [%s] to shared location [%s] using backup name [%s_",
+                        DEFAULT_CORE_NAME,
+                        backupCommand.backupLocation,
+                        DEFAULT_CORE_NAME)));
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format("Optimizing of collection [%s] is in progress.",
+                        DEFAULT_CORE_NAME)));
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format(
+                        "Backup failed. Unable to optimize collection [%s]",
+                        DEFAULT_CORE_NAME)));
+    }
+
+    /**
+     * Collections are optimized before backups. This test verifies that an error message is
+     * printed to the console when optimization of a collection throws an exception.
+     */
+    @Test
+    public void testSolrCloudBackupFailsDuringOptimizationThrowsException() throws Exception {
+        setupSolrClientType(SOLR_CLOUD_CLIENT_TYPE);
+
+        setupMockSolrClientForCollectionOptimizationThrowsException(DEFAULT_CORE_NAME);
+
+        BackupCommand backupCommand = getSynchronousBackupCommand(getBackupLocation(),
+                DEFAULT_CORE_NAME,
+                mockSolrClient);
+
+        backupCommand.doExecute();
+
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format(
+                        "Backing up collection [%s] to shared location [%s] using backup name [%s_",
+                        DEFAULT_CORE_NAME,
+                        backupCommand.backupLocation,
+                        DEFAULT_CORE_NAME)));
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format("Optimizing of collection [%s] is in progress.",
+                        DEFAULT_CORE_NAME)));
+        assertThat(consoleOutput.getOutput(),
+                containsString("Backup failed."));
```

Comment:
❓ For this test and `testSolrCloudBackupFailsDuringOptimizationWithErrorCode`, is there a way we can verify that the optimize call is actually made?